From 01beb00127c2ddae9e909fb87b8b7ce803ee3da4 Mon Sep 17 00:00:00 2001
From: Cai YiWei <cyw@rock-chips.com>
Date: Fri, 7 Sep 2018 14:55:04 +0800
Subject: [PATCH] media: rockchip/cif: fix cif restart not work

if stop cif timeout, stopping flag still true,
restart cif can't work.

Change-Id: If4c0805aa5d1231b1dab6046a826fe41d9796872
Signed-off-by: Cai YiWei <cyw@rock-chips.com>
---
 drivers/media/platform/rockchip/cif/capture.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/rockchip/cif/capture.c b/drivers/media/platform/rockchip/cif/capture.c
index d3619ad8cc5f..c3ebec506f02 100644
--- a/drivers/media/platform/rockchip/cif/capture.c
+++ b/drivers/media/platform/rockchip/cif/capture.c
@@ -362,8 +362,10 @@ static void rkcif_stop_streaming(struct vb2_queue *queue)
 	ret = wait_event_timeout(stream->wq_stopped,
 				 stream->state != RKCIF_STATE_STREAMING,
 				 msecs_to_jiffies(1000));
-	if (!ret)
+	if (!ret) {
 		rkcif_stream_stop(stream);
+		stream->stopping = false;
+	}
 	pm_runtime_put(dev->dev);
 
 	/* stop the sub device*/
-- 
2.35.3

